phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -160,6 +163,11 @@ public void preScannerOpen(org.apache.hadoop.hbase.coprocessor.ObserverContext<R
             TimeRange timeRange = scan.getTimeRange();
             scan.setTimeRange(timeRange.getMin(), Bytes.toLong(txnScn));
         }
+        byte[] cdcScan = scan.getAttribute(CDC_DATA_TABLE_DEF);
+        if (cdcScan != null) {
```

Comment:
Check if the scan is already adjusted (by checking if the scan is already changed to a raw scan.